.github/workflows/3rd-party-tests.yml

```
@@ -0,0 +1,88 @@
+name: Test on 3rd party libraries
+
+on:
+  push:
+    branches:
+      - "master"
+      - "sqlalchemy_pipeline"
+    paths-ignore:
+      - "docs/*"
+      - "tools/*"
+  workflow_dispatch:
+
+concurrency:
+  # Cancel older requests of the same workflow in the same branch.
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  sqlalchemy:
+    # linux should be enough to test if everything works.
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.11"
+        sqlalchemy:
+          - git+https://github.com/sqlalchemy/sqlalchemy.git#egg=SQLAlchemy
+          - sqlalchemy
+        impl:
+          - c
+          - python
+
+    env:
+      PSYCOPG_IMPL: ${{ matrix.impl }}
+      DEPS: ./psycopg[test] ./psycopg_pool pytest pytest-xdist
+
+    services:
+      pg:
+        image: postgres:15
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_DB: test
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup PG
+        env:
+          PGPASSWORD: password
+        run: |
+          run_sql="psql -h 127.0.0.1 -d test -U postgres -c "
+          $run_sql 'SELECT version();'
+          $run_sql 'CREATE SCHEMA test_schema; CREATE SCHEMA test_schema_2;CREATE EXTENSION hstore;'
```

They can be run all together in the same command. We can use -e to echo the commands run. I also prefer to specify a connection string instead of the loose parameters.

```suggestion
        run: |
          psql -AXqte "host=127.0.0.1 dbname=test user=postgres" << HERE
          SELECT version();
          CREATE SCHEMA test_schema;
          CREATE SCHEMA test_schema_2;
          CREATE EXTENSION hstore;
          HERE
```

This would produce an output like:

```
SELECT version();
PostgreSQL 15.1 (Ubuntu 15.1-1.pgdg22.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, 64-bit
CREATE SCHEMA test_schema;
CREATE SCHEMA test_schema_2;
CREATE EXTENSION hstore;
```

---

thanks, that looks better!